src/ot_http_status.erl

```diff
@@ -0,0 +1,106 @@
+%%%------------------------------------------------------------------------
+%% Copyright 2019, OpenTelemetry Authors
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc
+%% HTTP Status helper module. Provides functions to working with HTTP status
+%% codes to status records.
+%% @end
+%%%-------------------------------------------------------------------------
+-module(ot_http_status).
+
+-export([to_status/1,
+         to_status/2]).
+
+-include("opentelemetry.hrl").
+
+-type http_status_code() :: 1..599.
+
+-callback to_status(http_status_code()) -> opentelemetry:status().
+-callback to_status(http_status_code(), unicode:unicode_binary()) -> opentelemetry:status().
+
+-spec to_status(HttpStatusCode) -> opentelemetry:status() when
+      HttpStatusCode :: http_status_code().
+to_status(HttpStatusCode) ->
+    to_status(HttpStatusCode, <<"">>).
+
+-spec to_status(HttpStatusCode, Message) -> opentelemetry:status() when
+      HttpStatusCode :: http_status_code(),
+      Message :: unicode:unicode_binary().
+to_status(HttpStatusCode, Message) ->
+    Status =
+        case HttpStatusCode of
+            Code when Code >= 100 andalso Code < 300 ->
+                opentelemetry:status(?OTEL_STATUS_OK, Message);
```

Comment:
is this based off any spec? usually OK is for 200 and above, since 100-199 are about status over the connection (i.e. 100 Continue)

Reply:
There is. The spec is the table in the spec linked in the PR. It makes reasonable sense that those codes would correlate to a trace span's status being Ok. I don't know which of the available codes would make more sense, so I think they've done a decent job